jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -98,16 +100,30 @@ private Tracer(
     this.version = loadVersion();
 
     tags.put(Constants.JAEGER_CLIENT_VERSION_TAG_KEY, this.version);
-    String hostname = getHostName();
-    if (hostname != null) {
-      tags.put(Constants.TRACER_HOSTNAME_TAG_KEY, hostname);
+    Object hostNameTag = tags.get(Constants.TRACER_HOSTNAME_TAG_KEY);
+    if (hostNameTag == null) {
+      String hostname = getHostName();
+      if (hostname != null) {
+        tags.put(Constants.TRACER_HOSTNAME_TAG_KEY, hostname);
+      }
     }
-    int ipv4 ;
-    try {
-      tags.put(Constants.TRACER_IP_TAG_KEY, InetAddress.getLocalHost().getHostAddress());
-      ipv4 = Utils.ipToInt(Inet4Address.getLocalHost().getHostAddress());
-    } catch (UnknownHostException e) {
-      ipv4 = 0;
+    int ipv4;
+    Object ipTag = tags.get(Constants.TRACER_IP_TAG_KEY);
+    if (ipTag == null) {
+      try {
+        tags.put(Constants.TRACER_IP_TAG_KEY, InetAddress.getLocalHost().getHostAddress());
```

Comment:
can DRY `InetAddress.getLocalHost().getHostAddress()`

Reply:
Sorry, where am I repeating myself?

Reply:
sorry, nvm, the 2nd one is getting v4 address